File: UptimeAppIndicator.py
##!/usr/bin/env python

import gtk
import appindicator
import logging 
import os

# Set up logging
log_file = os.path.expanduser('~/.local/share/UptimeAppIndicator.log')
logging.basicConfig(filename=log_file, level=logging.ERROR, format='%(asctime)s %(message)s')

def get_uptime():
    try:
        with open('/proc/uptime', 'r') as f:
            uptime_seconds = float(f.readline().split()[0])
            hours, remainder = divmod(uptime_seconds, 3600)
            minutes, seconds = divmod(remainder, 60)
            uptime_string = "{:02}:{:02}".format(hours, minutes)
        return uptime_string
    except IOError:
        logging.error("Error: Unable to read /proc/uptime")
        return "Error: Unable to read /proc/uptime"
    except ValueError:
        logging.error("Error: Unable to parse uptime")
        return "Error: Unable to parse uptime"

def exit_item(_):
    gtk.main_quit()

if __name__ == "__main__":
    try:
        ind = appindicator.Indicator("uptime_indicator", "indicator-messages", appindicator.CATEGORY_APPLICATION_STATUS)
        ind.set_status(appindicator.STATUS_ACTIVE)
        ind.set_label(get_uptime(), "")

        # create a menu
        menu = gtk.Menu()

        # create some 
        exit = gtk.MenuItem('Exit')
        exit.connect('activate', exit_item)
        exit.show()
        menu.append(exit)

        ind.set_menu(menu)

        gtk.timeout_add(60000, lambda: ind.set_label(get_uptime(), ""))

        gtk.main()
    except ImportError:
        logging.error("Error: Unable to import necessary libaries")

File: UptimeAppIndicator.c
#include <gtk/gtk.h>
#include <libappindicator/app-indicator.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void get_uptime(char *uptime_str) {
    FILE *uptime_file = fopen("/proc/uptime", "r");
    if (uptime_file == NULL) {
        strcpy(uptime_str, "Error: Unable to read /proc/uptime");
        return;
    }

    double uptime_seconds;
    if (fscanf(uptime_file, "%lf", &uptime_seconds) != 1) {
        strcpy(uptime_str, "Error: Unable to parse uptime");
        return;
    }

    fclose(uptime_file);

    int hours = (int)uptime_seconds / 3600;
    int minutes = ((int)uptime_seconds % 3600) / 60;

    sprintf(uptime_str, "%02d:%02d", hours, minutes);
}

void exit_item(GtkWidget *widget, gpointer data) {
    gtk_main_quit();
}

int main(int argc, char **argv) {
    gtk_init(&argc, &argv);

    AppIndicator *ind = app_indicator_new("uptime_indicator", "indicator-messages", APP_INDICATOR_CATEGORY_APPLICATION_STATUS);
    app_indicator_set_status(ind, APP_INDICATOR_STATUS_ACTIVE);

    char uptime_str[100];
    get_uptime(uptime_str);
    app_indicator_set_label(ind, uptime_str, "");

    GtkWidget *menu = gtk_menu_new();

    GtkWidget *exit = gtk_menu_item_new_with_label("Exit");
    g_signal_connect(exit, "activate", G_CALLBACK(exit_item), NULL);
    gtk_widget_show(exit);
    gtk_menu_shell_append(GTK_MENU_SHELL(menu), exit);

    app_indicator_set_menu(ind, GTK_MENU(menu));

    g_timeout_add(60000, (GSourceFunc)get_uptime, uptime_str);

    gtk_main();

    return 0;
}